### ISSUE TITLE
Optimize Optitype resource usage

### ISSUE THREAD

**Issue by malachig** [griffithlab/analysis-wdls#56]
Currently for the optitype step we request 64GB of ram and say nothing about CPUs.  
https://github.com/griffithlab/analysis-wdls/blob/main/definitions/tools/optitype_dna.wdl

But when Cromwell requests 64GB of RAM from Google we get this reported back:

`"machineType": "custom-10-65536",`

```
Jul 06 12:02:22 malachi-immuno-test java[14290]: 2022-07-06 12:02:22,783 cromwell-system-akka.dispatchers.backend-dispatcher-8451 INFO  - PipelinesApiAsyncBackendJobExecutionActor [UUID(78967613)immuno.optitype:NA:2]: To comply with GCE custom machine requirements, cpu was adjusted from 1 to 10
```

This is an expensive machine that we then massively underutilize in the optitype script here:
https://github.com/genome/docker-immuno_tools-cwl/blob/master/optitype_script.sh

The bottom line is that we wind up with a machine with 64GB of RAM and 10 CPUs.  Its not clear if we need that much ram, but we never use more than 4 CPUs.  Thread usage for `sambamba` and `bwa mem` steps in the script are hard code to use 4 CPUs.

**Comment by malachig** [comment]
Let's parameterize both memory and CPU usage in the WDL, update optitype_script.sh to accept these parameters as input.  

Then update the sambamba and bwa mem steps to use these resources more reasonably. i.e. all available CPUs should be used for the `bwa mem` tasks and available RAM - 1GB (or 80% say) should be used for `sambamba`. 

This will involve updating the `tools/optitype_dna.wdl`, `optitype_script.sh`, and then rebuild the docker image, and test the workflow.  

In testing we should note the impact on total run time and cost.

**Comment by malachig** [comment]
In terms of the resources to actually use once we have the parameters to specify them, let's try using 8 CPUs and 32GB of RAM

**Comment by malachig** [comment]
I guess we will need to make any new parameters optional.  Otherwise this may break use of this tool in the CWL version of the pipeline:

https://github.com/genome/analysis-workflows/blob/master/definitions/tools/optitype_dna.cwl
